src/ParquetMsg.chpl

```diff
@@ -119,6 +124,21 @@ module ParquetMsg {
     return (subdoms, (+ reduce lengths));
   }
 
+  proc getRGSubdomains(bytesPerRG: [?D] ?t, maxRowGroups: int) {
+    var rgSubdomains: [D] [0..#maxRowGroups] domain(1);
+
+    var offset = 0;
+    for i in D {
+      for rg in 0..#maxRowGroups {
+        if bytesPerRG[i][rg] != 0 {
+          rgSubdomains[i][rg] = {offset..#bytesPerRG[i][rg]};
+          offset += bytesPerRG[i][rg];
```

Comment:
Well, since we are basing the next output on the result of the previous with `offset += bytesPerRG[i][rg];`, I don't think we can parallelize as-is. We could pull that into a precompute to parallelize, but this is such a small amount of execution time, I don't think it matters.

Reply:
oh! my bad I wasn't paying close enough attention! but yah ur right, it shouldn't matter either